### PR TITLE
#18834 Avoid mess sort order when try to save multiple fields

### PR DESCRIPTION
**Pull request by freddyucv**

### CODE REVIEW

File: dotCMS/src/integration-test/java/com/dotcms/datagen/FieldDataGen.java [hunk 1]
```diff
@@ -8,16 +8,14 @@
...
     private Class type = TextField.class;
     private boolean required = Boolean.FALSE;
-    private String velocityVarName = "testFieldVarname" + currentTime;
+    private String velocityVarName;
     private int sortOrder = 1;
-    private String values = "testValue" + currentTime;
-    private String hint = "testHint" + currentTime;
-    private String defaultValue = "testDefaultValue" + currentTime;
+    private String values;
+    private String hint;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/datagen/FieldDataGen.java:17]: Codacy found an issue: [Field hint has the same name as a method](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6326102)

File: dotCMS/src/main/java/com/dotcms/contenttype/business/FieldAPI.java [hunk 2]
```diff
@@ -128,7 +128,19 @@
...
+	 * Saves a new Field
+	 *
+	 * @param field
+	 * @param user
+	 * @param reorderIfNeed if it true then reorder all the fields,according to the field to save sort order
+	 * @return
+	 * @throws DotDataException
+	 * @throws DotSecurityException
+	 */
+	public Field save(Field field, User user, boolean reorderIfNeed) throws DotDataException, DotSecurityException;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/contenttype/business/FieldAPI.java:142]: Codacy found an issue: [Unnecessary modifier 'public' on method 'save': the method is declared in an interface type](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6326102)

File: dotCMS/src/integration-test/java/com/dotcms/datagen/FieldDataGen.java [hunk 3]
```diff
@@ -26,13 +24,29 @@
     private boolean unique = Boolean.FALSE;
     private String contentTypeId;
     private String relationType;
+    private String id;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/datagen/FieldDataGen.java:27]: Codacy found an issue: [Field id has the same name as a method](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6326102)

File: dotCMS/src/integration-test/java/com/dotcms/datagen/FieldDataGen.java [hunk 4]
```diff
@@ -8,16 +8,14 @@
...
 public class FieldDataGen extends AbstractDataGen<Field> {
 
-    private final long currentTime = System.currentTimeMillis();
-
-    private String name = "testFieldName" + currentTime;
+    private String name;
     private Class type = TextField.class;
     private boolean required = Boolean.FALSE;
-    private String velocityVarName = "testFieldVarname" + currentTime;
+    private String velocityVarName;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/datagen/FieldDataGen.java:14]: Codacy found an issue: [Field velocityVarName has the same name as a method](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6326102)

File: dotCMS/src/integration-test/java/com/dotcms/datagen/FieldDataGen.java (hunk 3 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/datagen/FieldDataGen.java:27]: Codacy found an issue: [Avoid variables with short names like id](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6326102)

File: dotCMS/src/integration-test/java/com/dotcms/contenttype/business/FieldAPIImplIntegrationTest.java [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+import com.dotcms.contenttype.model.field.RowField;
+import com.dotcms.contenttype.model.type.ContentType;
+import com.dotcms.datagen.ContentTypeDataGen;
+import com.dotcms.datagen.FieldDataGen;
+import com.dotcms.util.IntegrationTestInitService;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.liferay.portal.model.User;
+import io.vavr.Tuple2;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/contenttype/business/FieldAPIImplIntegrationTest.java:14]: Codacy found an issue: [Avoid unused imports such as 'io.vavr.Tuple2'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6326102)

File: dotCMS/src/integration-test/java/com/dotcms/datagen/FieldDataGen.java [hunk 6]
```diff
@@ -8,16 +8,14 @@
 
 public class FieldDataGen extends AbstractDataGen<Field> {
 
-    private final long currentTime = System.currentTimeMillis();
-
-    private String name = "testFieldName" + currentTime;
+    private String name;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/datagen/FieldDataGen.java:11]: Codacy found an issue: [Field name has the same name as a method](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6326102)

File: dotCMS/src/integration-test/java/com/dotcms/contenttype/business/FieldAPIImplIntegrationTest.java [hunk 7]
```diff
@@ -0,0 +1,172 @@
...
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static com.dotcms.util.CollectionsUtils.list;
+import static org.junit.Assert.assertEquals;
+
+public class FieldAPIImplIntegrationTest {
+
+    @BeforeClass
+    public static void prepare () throws Exception {
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/contenttype/business/FieldAPIImplIntegrationTest.java:27]: Codacy found an issue: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6326102)

File: dotCMS/src/integration-test/java/com/dotcms/datagen/FieldDataGen.java [hunk 8]
```diff
@@ -26,13 +24,29 @@
     private boolean unique = Boolean.FALSE;
     private String contentTypeId;
     private String relationType;
+    private String id;
+
+    public FieldDataGen() {
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/datagen/FieldDataGen.java:29]: Codacy found an issue: [It is a good practice to call super() in a constructor](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6326102)

File: dotCMS/src/integration-test/java/com/dotcms/datagen/FieldDataGen.java [hunk 9]
```diff
@@ -26,13 +24,29 @@
...
+        defaultValue = "testDefaultValue" + currentTime;
+    }
 
     @SuppressWarnings("unused")
     public FieldDataGen name(final String name) {
         this.name = name;
         return this;
     }
 
+    public FieldDataGen id(final String id) {
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/datagen/FieldDataGen.java:45]: Codacy found an issue: [Avoid variables with short names like id](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6326102)

Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/datagen/FieldDataGen.java:45]: Codacy found an issue: [Avoid using short method names](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6326102)

File: dotCMS/src/integration-test/java/com/dotcms/datagen/FieldDataGen.java (hunk 3 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/datagen/FieldDataGen.java:27]: Codacy found an issue: [Avoid variables with short names like id](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6326102)

File: dotCMS/src/integration-test/java/com/dotcms/contenttype/business/FieldAPIImplIntegrationTest.java (hunk 5 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/contenttype/business/FieldAPIImplIntegrationTest.java:14]: Codacy found an issue: [Avoid unused imports such as 'io.vavr.Tuple2'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6326102)

File: dotCMS/src/integration-test/java/com/dotcms/datagen/FieldDataGen.java (hunk 8 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/datagen/FieldDataGen.java:29]: Codacy found an issue: [It is a good practice to call super() in a constructor](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6326102)

File: dotCMS/src/integration-test/java/com/dotcms/datagen/FieldDataGen.java (hunk 1 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/datagen/FieldDataGen.java:17]: Codacy found an issue: [Field hint has the same name as a method](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6326102)

File: dotCMS/src/integration-test/java/com/dotcms/datagen/FieldDataGen.java (hunk 3 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/datagen/FieldDataGen.java:27]: Codacy found an issue: [Field id has the same name as a method](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6326102)

File: dotCMS/src/integration-test/java/com/dotcms/datagen/FieldDataGen.java (hunk 9 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/datagen/FieldDataGen.java:45]: Codacy found an issue: [Avoid variables with short names like id](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6326102)

File: dotCMS/src/integration-test/java/com/dotcms/datagen/FieldDataGen.java (hunk 6 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/datagen/FieldDataGen.java:11]: Codacy found an issue: [Field name has the same name as a method](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6326102)

File: dotCMS/src/integration-test/java/com/dotcms/datagen/FieldDataGen.java (hunk 4 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/datagen/FieldDataGen.java:14]: Codacy found an issue: [Field velocityVarName has the same name as a method](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6326102)

File: dotCMS/src/integration-test/java/com/dotcms/contenttype/business/FieldAPIImplIntegrationTest.java (hunk 7 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/contenttype/business/FieldAPIImplIntegrationTest.java:27]: Codacy found an issue: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6326102)

File: dotCMS/src/main/java/com/dotcms/contenttype/business/FieldAPI.java (hunk 2 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/contenttype/business/FieldAPI.java:142]: Codacy found an issue: [Unnecessary modifier 'public' on method 'save': the method is declared in an interface type](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6326102)

File: dotCMS/src/integration-test/java/com/dotcms/datagen/FieldDataGen.java (hunk 9 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/datagen/FieldDataGen.java:45]: Codacy found an issue: [Avoid using short method names](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6326102)

File: dotCMS/src/main/java/com/dotcms/contenttype/business/FieldAPI.java [hunk 10]
```diff
@@ -128,7 +128,19 @@
...
 	 */
 	Field save(Field field, User user) throws DotDataException, DotSecurityException;
-	
+
+	/**
+	 * Saves a new Field
+	 *
+	 * @param field
+	 * @param user
+	 * @param reorderIfNeed if it true then reorder all the fields,according to the field to save sort order
```
Comment (dsilvam) [inline dotCMS/src/main/java/com/dotcms/contenttype/business/FieldAPI.java:137]: let's rephrase this

**Review by dsilvam — APPROVED** [review]

File: dotCMS/src/integration-test/java/com/dotcms/datagen/FieldDataGen.java (hunk 9 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/datagen/FieldDataGen.java:45]: Codacy found an issue: [Avoid using short method names](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6326102)

File: dotCMS/src/integration-test/java/com/dotcms/datagen/FieldDataGen.java (hunk 6 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/datagen/FieldDataGen.java:11]: Codacy found an issue: [Field name has the same name as a method](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6326102)

File: dotCMS/src/integration-test/java/com/dotcms/datagen/FieldDataGen.java (hunk 1 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/datagen/FieldDataGen.java:17]: Codacy found an issue: [Field hint has the same name as a method](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6326102)

File: dotCMS/src/integration-test/java/com/dotcms/datagen/FieldDataGen.java (hunk 3 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/datagen/FieldDataGen.java:27]: Codacy found an issue: [Field id has the same name as a method](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6326102)

File: dotCMS/src/integration-test/java/com/dotcms/datagen/FieldDataGen.java (hunk 4 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/datagen/FieldDataGen.java:14]: Codacy found an issue: [Field velocityVarName has the same name as a method](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6326102)

File: dotCMS/src/integration-test/java/com/dotcms/contenttype/business/FieldAPIImplIntegrationTest.java (hunk 7 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/contenttype/business/FieldAPIImplIntegrationTest.java:27]: Codacy found an issue: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6326102)

File: dotCMS/src/integration-test/java/com/dotcms/datagen/FieldDataGen.java (hunk 9 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/datagen/FieldDataGen.java:45]: Codacy found an issue: [Avoid variables with short names like id](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6326102)

File: dotCMS/src/integration-test/java/com/dotcms/datagen/FieldDataGen.java (hunk 3 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/datagen/FieldDataGen.java:27]: Codacy found an issue: [Avoid variables with short names like id](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6326102)

File: dotCMS/src/main/java/com/dotcms/contenttype/business/FieldAPI.java [hunk 11]
```diff
@@ -128,7 +128,19 @@
...
+	 * Saves a new Field
+	 *
+	 * @param field
+	 * @param user
+	 * @param reorderIfNeed if it's true then reorder all the fields relative to the order of the field being saved
+	 * @return
+	 * @throws DotDataException
+	 * @throws DotSecurityException
+	 */
+	public Field save(Field field, User user, boolean reorderIfNeed) throws DotDataException, DotSecurityException;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/contenttype/business/FieldAPI.java:142]: Codacy found an issue: [Unnecessary modifier 'public' on method 'save': the method is declared in an interface type](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6326102)

File: dotCMS/src/integration-test/java/com/dotcms/contenttype/business/FieldAPIImplIntegrationTest.java (hunk 5 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/contenttype/business/FieldAPIImplIntegrationTest.java:14]: Codacy found an issue: [Avoid unused imports such as 'io.vavr.Tuple2'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6326102)

File: dotCMS/src/integration-test/java/com/dotcms/datagen/FieldDataGen.java (hunk 8 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/datagen/FieldDataGen.java:29]: Codacy found an issue: [It is a good practice to call super() in a constructor](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6326102)

Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/datagen/FieldDataGen.java:29]: Codacy found an issue: [It is a good practice to call super() in a constructor](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6326102)

File: dotCMS/src/integration-test/java/com/dotcms/datagen/FieldDataGen.java (hunk 4 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/datagen/FieldDataGen.java:14]: Codacy found an issue: [Field velocityVarName has the same name as a method](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6326102)

File: dotCMS/src/integration-test/java/com/dotcms/datagen/FieldDataGen.java (hunk 9 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/datagen/FieldDataGen.java:45]: Codacy found an issue: [Avoid using short method names](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6326102)

Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/datagen/FieldDataGen.java:45]: Codacy found an issue: [Avoid variables with short names like id](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6326102)

File: dotCMS/src/integration-test/java/com/dotcms/datagen/FieldDataGen.java (hunk 3 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/datagen/FieldDataGen.java:27]: Codacy found an issue: [Avoid variables with short names like id](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6326102)

File: dotCMS/src/integration-test/java/com/dotcms/contenttype/business/FieldAPIImplIntegrationTest.java [hunk 12]
```diff
@@ -0,0 +1,171 @@
...
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/contenttype/business/FieldAPIImplIntegrationTest.java:26]: Codacy found an issue: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6326102)

File: dotCMS/src/integration-test/java/com/dotcms/datagen/FieldDataGen.java (hunk 3 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/datagen/FieldDataGen.java:27]: Codacy found an issue: [Field id has the same name as a method](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6326102)

File: dotCMS/src/integration-test/java/com/dotcms/datagen/FieldDataGen.java (hunk 1 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/datagen/FieldDataGen.java:17]: Codacy found an issue: [Field hint has the same name as a method](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6326102)

File: dotCMS/src/integration-test/java/com/dotcms/datagen/FieldDataGen.java (hunk 6 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/datagen/FieldDataGen.java:11]: Codacy found an issue: [Field name has the same name as a method](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6326102)

**Review by dsilvam — APPROVED** [review]

**Review by jdotcms — APPROVED** [review]

File: dotCMS/src/integration-test/java/com/dotcms/datagen/FieldDataGen.java (hunk 1 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/datagen/FieldDataGen.java:17]: Codacy found an issue: [Field hint has the same name as a method](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6326102)

File: dotCMS/src/integration-test/java/com/dotcms/datagen/FieldDataGen.java (hunk 9 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/datagen/FieldDataGen.java:45]: Codacy found an issue: [Avoid using short method names](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6326102)

File: dotCMS/src/integration-test/java/com/dotcms/datagen/FieldDataGen.java (hunk 3 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/datagen/FieldDataGen.java:27]: Codacy found an issue: [Field id has the same name as a method](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6326102)

File: dotCMS/src/integration-test/java/com/dotcms/datagen/FieldDataGen.java (hunk 6 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/datagen/FieldDataGen.java:11]: Codacy found an issue: [Field name has the same name as a method](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6326102)

File: dotCMS/src/integration-test/java/com/dotcms/contenttype/business/FieldAPIImplIntegrationTest.java (hunk 12 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/contenttype/business/FieldAPIImplIntegrationTest.java:26]: Codacy found an issue: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6326102)

File: dotCMS/src/integration-test/java/com/dotcms/datagen/FieldDataGen.java (hunk 4 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/datagen/FieldDataGen.java:14]: Codacy found an issue: [Field velocityVarName has the same name as a method](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6326102)

File: dotCMS/src/integration-test/java/com/dotcms/datagen/FieldDataGen.java (hunk 9 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/datagen/FieldDataGen.java:45]: Codacy found an issue: [Avoid variables with short names like id](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6326102)

File: dotCMS/src/integration-test/java/com/dotcms/datagen/FieldDataGen.java (hunk 8 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/datagen/FieldDataGen.java:29]: Codacy found an issue: [It is a good practice to call super() in a constructor](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6326102)

File: dotCMS/src/integration-test/java/com/dotcms/datagen/FieldDataGen.java (hunk 3 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/datagen/FieldDataGen.java:27]: Codacy found an issue: [Avoid variables with short names like id](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6326102)

**Pull request merged**